src/reducers/versions.tsx

```diff
@@ -687,7 +687,9 @@ export const fetchDiff = ({
       dispatch(errorsActions.addError({ error: response.error }));
       dispatch(actions.abortFetchDiff());
     } else {
-      dispatch(actions.loadVersionInfo({ version: response }));
+      if (!getVersionInfo(versionsState, response.id)) {
+        dispatch(actions.loadVersionInfo({ version: response }));
```

Comment:
This seems fine as a short term fix but it doesn't seem like the right thing to do. It's better than the last iteration but it still would never allow us to reload a version from the server when calling `fetchDiff()`. I suggest filing an issue to introduce another action, like `fetchDiffPath()`, which would only be for switching paths within the same version. That seems like a better way to fix it. 

Alternatively, we could introduce a new action parameter like `loadVersion: boolean` to indicate whether a version should be loaded.

Reply:
Thanks @kumar303. I filed #671